### PR TITLE
add module test

### PR DESCRIPTION
**Pull request by chenyang23333**

### CODE REVIEW

File: examples/stable_diffusion_xl/tests/st/BasicTransformerBlockFA_forward.py [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+            use_scale_shift_norm=use_scale_shift_norm,
+        )
+        net = NetWithLoss(_net)
+        net = auto_mixed_precision(net, amp_level="O0")
+        input_size = ((1, model_channels, 128, 128), (1, 1280))  # x, emb
+        dataset_column_names = ["data1", "data2"]
+    elif args.net == "MemoryEfficientCrossAttention":
+        from gm.modules.attention import MemoryEfficientCrossAttention
+        from gm.util.util import auto_mixed_precision
+
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tests/st/BasicTransformerBlockFA_forward.py:125]: 每个文件仅保留自己的分支

File: examples/stable_diffusion_xl/tests/st/test_module.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Test_module
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tests/st/test_module.md:1]: 1、/path to/st -> /path to/test_modules
2、test_module.md -> README.md

File: examples/stable_diffusion_xl/tests/test_modules/BasicTransformerBlock_forward.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    ds = ds.batch(
+        per_batch_size,
+        drop_remainder=drop_remainder,
+    )
+    ds = ds.repeat(1)
+
+    return ds
+
+
+class Dataset:
```
Comment (SamitHuang) [inline examples/stable_diffusion_xl/tests/test_modules/BasicTransformerBlock_forward.py:44]: Too many duplicated class implementation for Dataset. 
For those classes and functions are commonly used by all test scripts, please put them into a file like  `common.py`

Comment (chenyang23333) [reply examples/stable_diffusion_xl/tests/test_modules/BasicTransformerBlock_forward.py]: Done

**Review by SamitHuang — APPROVED** [review]

**Review by zhanghuiyao — APPROVED** [review]

**Pull request merged**